display_list/dl_color.h

```diff
@@ -41,7 +41,10 @@ struct DlColor {
         red_(red),
         green_(green),
         blue_(blue),
-        color_space_(colorspace) {}
+        color_space_(colorspace) {
+    FML_DCHECK(alpha >= 0.0);
+    FML_DCHECK(alpha <= 1.0);
```

Comment:
Should we dcheck or clamp? This is currently, in this PR, helping to catch a problem in a test conversion that we crossed paths with, but that bug-in-a-test would have normally been caught by the test creator. If we dcheck then we might annoy someone who calculated the alpha and ended up with a barely-out-of-range value. They could just have easily have provided 255 for one of the color values and we'd take it as a potential non-normalized Color Space float...?

Reply:
Rather than dcheck I think we should clamp right? Certain blend modes (plus) can lead to alpha greater than one without a clamp

Reply:
> Rather than dcheck I think we should clamp right? Certain blend modes (plus) can lead to alpha greater than one without a clamp

My concern with that is that it will mask problems like we had where someone is sending in 255 to create a color.  The assert forces people creating colors to have a clamp if they are doing arithmetic that will get it out of bounds.  I didn't have to change any code as a result of all our tests, so that's a good sign.  I think this leaves us in a better position considering we control all the locations where DlColors are created.  A public API would be a different story.

Reply:
You should clamp.

Reply:
I removed the asserts to duplicate the behavior of `impeller::Color`.  There may be transitory colors that are not valid, clamping or the asserts could mess with that.  I don't think we blend with DlColor today, it's more of a transmission format, but I think we plan to use it more broadly in the future.

Reply:
We will be CPU blending once we merge DlColor and impeller::Color. This will happen sooner than you'd think